service/history/workflow/delete_manager.go

```diff
@@ -65,20 +70,57 @@ func NewDeleteManager(
 	cache Cache,
 	config *configs.Config,
 	archiverClient archiver.Client,
+	timeSource clock.TimeSource,
 ) *DeleteManagerImpl {
 	deleteManager := &DeleteManagerImpl{
 		shard:          shard,
 		historyCache:   cache,
 		metricsClient:  shard.GetMetricsClient(),
 		config:         config,
 		archivalClient: archiverClient,
+		timeSource:     timeSource,
 	}
 
 	return deleteManager
 }
+func (m *DeleteManagerImpl) AddDeleteWorkflowExecutionTask(
+	nsID namespace.ID,
+	we commonpb.WorkflowExecution,
+	ms MutableState,
+) error {
+
+	if ms.IsWorkflowExecutionRunning() {
+		// Running workflow cannot be deleted. Close or terminate it first.
+		return consts.ErrWorkflowNotCompleted
+	}
+
+	taskGenerator := NewTaskGenerator(
+		m.shard.GetNamespaceRegistry(),
+		ms,
+	)
```

Comment:
cc @yycptt

Reply:
Ack. Will update this one as well when injecting task generator.